PYTHON-5111 Update datetime_conversion in docstrings of MongoClients

caseyclements opened the pull request:
Reflects the fact that it is now in kwargs.

ShaneHarvey commented:
![Screenshot 2025-02-06 at 12 00 31 PM](https://github.com/user-attachments/assets/e37c9978-041a-4a42-ab0a-6e733b20f687)
The format looks a little off. Perhaps we need a colon at the end? And could we bold the line?

caseyclements commented:
The format was off before. The text in the last :param gets indented. It's weird, but  I played with it to make it look pretty decent, adding kwargs as a param, and then having the list below.

<img width="782" alt="image" src="https://github.com/user-attachments/assets/5bd52a8a-4632-4b94-8c2f-49e47f4fe4ca" />

ShaneHarvey reviewed (changes requested):
There have been no code changes since my last review.

caseyclements commented:
> There have been no code changes since my last review.

That's right. What I am saying is that Sphinx adds a carriage return to the last :param. This was how it was before, too. My picture shows the docs as they have existing for however long. When I moved datetime_conversion down, this weird behavior appears for :param type_registry.  This is why I added :param kwargs, specifically because it included a big indented list before. There previously was a bold line, but it was redundant if any text was added to kwargs. 

Suggestions are welcome, given the constraint. I'm pretty happy with how it looks now.

ShaneHarvey commented:
Oh I get it now. Could we at least add a colon and bold "Additional optional parameters available as keyword arguments:". That would better match with the other sections like "Write Concern options:"

caseyclements commented:
@ShaneHarvey Done. Bold with colon.
